### PR TITLE
Be aware of opt-in full replace-by-fee.

### PR DESCRIPTION
**Pull request by schildbach**

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/TransactionInput.java [hunk 1]
```diff
@@ -46,9 +48,7 @@
     // Magic outpoint index that indicates the input is in fact unconnected.
     private static final long UNCONNECTED = 0xFFFFFFFFL;
 
-    // Allows for altering transactions after they were broadcast. Tx replacement is currently disabled in the C++
-    // client so this is always the UINT_MAX.
-    // TODO: Document this in more detail and build features that use it.
+    // Allows for altering transactions after they were broadcast. Values below NO_SEQUENCE mean it can be altered.
```
Comment (jlopp) [inline core/src/main/java/org/bitcoinj/core/TransactionInput.java]: values below NO_SEQUENCE - 1, right?

**Comment by jlopp** [comment]
utACK

File: core/src/main/java/org/bitcoinj/core/TransactionInput.java (hunk 1 above)
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/TransactionInput.java]: Yes, that's right. I was tying to describe the original meaning of the sequence. I'll amend.

Do you know why BIP 125 leaves this "gap" at NO_SEQUENCE-1?

Comment (jlopp) [inline core/src/main/java/org/bitcoinj/core/TransactionInput.java]: I believe that the idea is that:

nSequence == 0xffffffff disables nLockTime field values & finalizes a txn
nSequence == 0xffffffff - 1 explicitly disables RBF while still allowing nLockTime field values

Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/TransactionInput.java]: Makes sense. Thanks for the review!

**Pull request merged**